frontend/app/login/components/__tests__/LoginForm.test.tsx

```diff
@@ -1,11 +1,27 @@
 import { render, screen, userEvent, waitFor, setupMockServer } from 'lib/jest'
 import { LoginForm } from '../LoginForm'
 import { faker } from '@faker-js/faker'
-import { apiLoginHandler } from 'mocks/login-api-handler'
 import { db } from 'mocks/db'
 
-// ? Setup server endpoint
-setupMockServer(apiLoginHandler)
+jest.mock('next-auth/react', () => {
+  return {
+    signIn: (_type: string, credentials: { username: string; password: string }) => {
+      const { username, password } = credentials
+      const user = db.user.findFirst({
+        where: {
+          username: {
+            equals: username
+          },
+          password: {
+            equals: password
+          }
+        }
+      })
```

Comment:
can you explain what does `mocks/db` do? Why do you need this?

Reply:
it's a factory 🏭 . Just want to initialize the data inside the app in order to move around and test with the expected output.

```js
import { factory, primaryKey } from '@mswjs/data'
import { faker } from '@faker-js/faker'

export const db = factory({
  // Create a "user" model,
  user: {
    id: primaryKey(faker.datatype.uuid),
    username: () => faker.internet.userName(),
    password: () => faker.internet.password()
  }
})

```

So in this case, created a user before all the tests then used that user info through out the login test

Reply:
Oooh interesting. Ok gotcha!